### PR TITLE
[Modules] Feature/azure firewall basic SKU

### LINKED ISSUE #3305: [Feature Request]: azure firewall: support for 'Basic' sku
### Description

There is a new Azure FW sku 'Basic' since a while, long awaited as the existing skus where too expensive for some scenarios.

Would be great if you added support for Azure FW Basic.

### PR DESCRIPTION
**Pull request by DanoThom**
# Description
Fixes #3305 

Adds the newer 'Basic' SKU for Azure Firewall and it's accompanying Management Public IP logic.

> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

| Pipeline |
| - |
| |

# Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

File: modules/network/azure-firewalls/main.bicep [hunk 1]
```diff
@@ -244,6 +309,7 @@ resource azureFirewall 'Microsoft.Network/azureFirewalls@2022-07-01' = {
       id: firewallPolicyId
     } : null
     ipConfigurations: ipConfigurations
+    managementIpConfiguration: requiresManagementIp == true ? managementIPConfiguration : null
```
Comment (AlexanderSehr) [inline modules/network/azure-firewalls/main.bicep:312]:
```suggestion
    managementIpConfiguration: requiresManagementIp ? managementIPConfiguration : null
```

File: modules/network/azure-firewalls/main.bicep [hunk 2]
```diff
@@ -115,6 +122,16 @@ param diagnosticMetricsToEnable array = [
 @description('Optional. The name of the diagnostic setting, if deployed. If left empty, it defaults to "<resourceName>-diagnosticSettings".')
 param diagnosticSettingsName string = ''
 
+var azureSkuName = empty(vNetId) ? 'AZFW_Hub' : 'AZFW_VNet'
+var requiresManagementIp = azureSkuTier == 'Basic' ? true : false
+var isCreateDefaultManagementIP = empty(managementIPResourceID) && requiresManagementIp == true
```
Comment (AlexanderSehr) [inline modules/network/azure-firewalls/main.bicep:127]:
```suggestion
var isCreateDefaultManagementIP = empty(managementIPResourceID) && requiresManagementIp
```

File: modules/network/azure-firewalls/main.bicep [hunk 3]
```diff
@@ -158,6 +165,33 @@ var ipConfigurations = concat([
...
+    id: '${vNetId}/subnets/AzureFirewallManagementSubnet' // The subnet name must be AzureFirewallManagementSubnet for a 'Basic' SKU tier firewall
+  }
+}
+var existingMip = {
+  publicIPAddress: {
+    id: managementIPResourceID
+  }
+}
+var newMip = {
+  publicIPAddress: (empty(managementIPResourceID) && isCreateDefaultManagementIP) ? {
```
Comment (AlexanderSehr) [inline modules/network/azure-firewalls/main.bicep:183]:
```suggestion
  publicIPAddress: empty(managementIPResourceID) && isCreateDefaultManagementIP ? {
```

**Review by AlexanderSehr — DISMISSED** [review]

**Comment by AlexanderSehr** [comment]
@DanoThom did you successfully run the test files in your environment - or don't you have an environment available? If not, I would suggest we merge your PR into a branch and test it on our end.

**Comment by DanoThom** [comment]
> @DanoThom did you successfully run the test files in your environment - or don't you have an environment available? If not, I would suggest we merge your PR into a branch and test it on our end.

@AlexanderSehr no I don't have an environment and was unable to run the tests. I have pushed your suggested changes though.

By the way, in trying to keep things in sync, I just did a local merge of main into my my branch, successfully resolved a conflict, but it also resulted in about 2 thousand unexpected additional changed files?! Has this repo been under a lot of churn lately?
In this situation, what's the best option - a rebase onto main?

**Comment by AlexanderSehr** [comment]
> > @DanoThom did you successfully run the test files in your environment - or don't you have an environment available? If not, I would suggest we merge your PR into a branch and test it on our end.
> 
> @AlexanderSehr no I don't have an environment and was unable to run the tests. I have pushed your suggested changes though.
> 
> By the way, in trying to keep things in sync, I just did a local merge of main into my my branch, successfully resolved a conflict, but it also resulted in about 2 thousand unexpected additional changed files?! Has this repo been under a lot of churn lately? In this situation, what's the best option - a rebase onto main?

Hey @DanoThom thanks for the changes. For some time now we're preparing the modules to be merged into the official [Public Bicep Registry](https://github.com/Azure/bicep-registry-modules) which unfortunately requires us to change a lot of things we're tracking in several issues such as #3131. Not exactly a fun exercise which results in many changes due to changing paths etc. - which is in turn also the reason we have not released version 0.11.0 yet - but are almost there. I'd suggest to wait for the remaining changes to be merged first and then rebase.

I'll go ahead and change your target branch to a temp one on our side and will run the test pipeline from there.

**Pull request merged**